From 212ff0328363476c794070330b3e4515399ddb71 Mon Sep 17 00:00:00 2001
From: Mihail Konev <k.mvc@ya.ru>
Date: Sun, 5 Mar 2017 22:58:09 +0500
Subject: [PATCH 2/2] fix current dir inheritance for alt-f2 on msys2

---
 src/child.c   |  3 +++
 src/winmain.c | 14 ++++++++++++++
 2 files changed, 17 insertions(+)

diff --git a/src/child.c b/src/child.c
index f1fee3c..934ca9c 100644
--- a/src/child.c
+++ b/src/child.c
@@ -1189,6 +1189,9 @@ do_child_fork(int argc, char *argv[], int moni, bool launch, bool config_size, b
     if (icon_is_from_shortcut)
       setenv("MINTTY_ICON", cs__wcstoutf(cfg.icon), true);
 
+    // tell alt-f2 mintty instance that it is such one
+    setenv("MINTTY_IS_CHILD", "1", 1);
+
     //setenv("MINTTY_CHILD", "1", true);
 
 #if CYGWIN_VERSION_DLL_MAJOR >= 1005
diff --git a/src/winmain.c b/src/winmain.c
index 9e95197..7b54043 100644
--- a/src/winmain.c
+++ b/src/winmain.c
@@ -4782,6 +4782,12 @@ opts[] = {
   {0, 0, 0, 0}
 };
 
+static int getenvi(const char *varname) {
+  char *s = getenv(varname);
+  if (s == NULL) { return 0; }
+  return atoi(s);
+}
+
 int
 main(int argc, char *argv[])
 {
@@ -6001,6 +6007,14 @@ static int dynfonts = 0;
     launcher_setup_argv();
   }
 
+  // Ask /etc/post-install/05-home-dir.post not to "cd ~/" in an Alt-F2 window
+  {
+    if (getenvi("MINTTY_IS_CHILD")) {
+      putenv("CHERE_INVOKING=1");
+    }
+    unsetenv("MINTTY_IS_CHILD");
+  }
+
   // Create child process.
   /* We could move this below SetFocus() or win_init_drop_target() 
      in order to further reduce the delay until window display (#1113) 
-- 
2.32.0

