From 3b39dbe8bf6b8bd45200d93df17bd06fd109fb11 Mon Sep 17 00:00:00 2001
From: Mihail Konev <k.mvc@ya.ru>
Date: Sun, 5 Mar 2017 22:58:09 +0500
Subject: [PATCH 2/2] fix current dir inheritance for alt-f2 on msys2

---
 src/child.c   |  3 +++
 src/winmain.c | 13 +++++++++++++
 2 files changed, 16 insertions(+)

diff --git a/src/child.c b/src/child.c
index 06ef1da..d99cd06 100644
--- a/src/child.c
+++ b/src/child.c
@@ -1163,6 +1163,9 @@ do_child_fork(int argc, char *argv[], int moni, bool launch, bool config_size)
     if (icon_is_from_shortcut)
       setenv("MINTTY_ICON", cs__wcstoutf(cfg.icon), true);
 
+    // tell alt-f2 mintty instance that it is such one
+    setenv("MINTTY_IS_CHILD", "1", 1);
+
     //setenv("MINTTY_CHILD", "1", true);
 
 #if CYGWIN_VERSION_DLL_MAJOR >= 1005
diff --git a/src/winmain.c b/src/winmain.c
index d6516de..3f925ec 100644
--- a/src/winmain.c
+++ b/src/winmain.c
@@ -4488,6 +4488,12 @@ opts[] = {
   {0, 0, 0, 0}
 };
 
+static int getenvi(const char *varname) {
+  char *s = getenv(varname);
+  if (s == NULL) { return 0; }
+  return atoi(s);
+}
+
 int
 main(int argc, char *argv[])
 {
@@ -5643,6 +5649,13 @@ typedef UINT_PTR uintptr_t;
       launcher_setup_env();
       launcher_setup_argv();
     }
+    // Ask /etc/post-install/05-home-dir.post not to "cd ~/" in an Alt-F2 window
+    {
+      if (getenvi("MINTTY_IS_CHILD")) {
+        putenv("CHERE_INVOKING=1");
+      }
+      unsetenv("MINTTY_IS_CHILD");
+    }
 
     // Create child process.
     child_create(
-- 
2.28.0

