From 60901af858c227a128a00a6355bc45101ee5ac27 Mon Sep 17 00:00:00 2001
From: Mihail Konev <k.mvc@ya.ru>
Date: Sun, 5 Mar 2017 22:58:09 +0500
Subject: [PATCH 2/3] fix current dir inheritance for alt-f2 on msys2

---
 src/child.c   |  2 ++
 src/winmain.c | 13 +++++++++++++
 2 files changed, 15 insertions(+)

diff --git a/src/child.c b/src/child.c
index cd6bf00..0189ab1 100644
--- a/src/child.c
+++ b/src/child.c
@@ -1137,6 +1137,8 @@ do_child_fork(int argc, char *argv[], int moni, bool launch)
     // propagate shortcut-inherited icon
     if (icon_is_from_shortcut)
       setenv("MINTTY_ICON", cs__wcstoutf(cfg.icon), true);
+    // tell alt-f2 mintty instance that it is such one
+    setenv("MINTTY_IS_CHILD", "1", 1);
 
     //setenv("MINTTY_CHILD", "1", true);
 
diff --git a/src/winmain.c b/src/winmain.c
index 89a6b3b..726b5ba 100644
--- a/src/winmain.c
+++ b/src/winmain.c
@@ -3914,6 +3914,12 @@ opts[] = {
   {0, 0, 0, 0}
 };
 
+static int getenvi(const char *varname) {
+  char *s = getenv(varname);
+  if (s == NULL) { return 0; }
+  return atoi(s);
+}
+
 int
 main(int argc, char *argv[])
 {
@@ -4945,6 +4951,13 @@ main(int argc, char *argv[])
       launcher_setup_env();
       launcher_setup_argv();
     }
+    // Ask /etc/post-install/05-home-dir.post not to "cd ~/" in an Alt-F2 window
+    {
+      if (getenvi("MINTTY_IS_CHILD")) {
+        putenv("CHERE_INVOKING=1");
+      }
+      unsetenv("MINTTY_IS_CHILD");
+    }
 
     // Create child process.
     child_create(
-- 
2.22.0

