From 2506d2be13a0fe6889d0f815ea9192d7cb6e54b6 Mon Sep 17 00:00:00 2001
From: mintty <mintty@users.noreply.github.com>
Date: Sat, 6 Feb 2021 00:00:00 +0100
Subject: [PATCH] safeguard

---
 src/winmain.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/winmain.c b/src/winmain.c
index cccc0d0b..6e69594e 100644
--- a/src/winmain.c
+++ b/src/winmain.c
@@ -700,6 +700,7 @@ win_set_title(char *title)
       GetWindowTextW(wnd, oldtitle, len + 1);
       if (0 != wcscmp(wtitle, oldtitle)) {
         SetWindowTextW(wnd, wtitle);
+        usleep(1000);
         update_tab_titles();
       }
     }
