diff --git a/src/hid_win.c b/src/hid_win.c
index 018b4d9..d56000a 100644
--- a/src/hid_win.c
+++ b/src/hid_win.c
@@ -18,7 +18,7 @@
 #include <devpropdef.h>
 #include <hidclass.h>
 #include <hidsdi.h>
-
+#include <wchar.h>
 #include "fido.h"
 
 #if defined(__MINGW32__) &&  __MINGW64_VERSION_MAJOR < 6
@@ -27,10 +27,8 @@ WINSETUPAPI WINBOOL WINAPI SetupDiGetDevicePropertyW(HDEVINFO,
     DWORD, PDWORD, DWORD);
 #endif
 
-#if defined(__MINGW32__)
 DEFINE_DEVPROPKEY(DEVPKEY_Device_Parent, 0x4340a6c5, 0x93fa, 0x4706, 0x97,
     0x2c, 0x7b, 0x64, 0x80, 0x08, 0xa5, 0xa7, 8);
-#endif
 
 struct hid_win {
 	HANDLE		dev;
