From 43f4480460bad913f00c8b401eebc5ac8a965d15 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Tue, 14 Feb 2017 14:45:21 +0300
Subject: [PATCH 1/7] perl.cygwin-auto-image-base

---
 Makefile.SH | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.SH b/Makefile.SH
index 2895bfc..b5bc906 100755
--- a/Makefile.SH
+++ b/Makefile.SH
@@ -79,7 +79,7 @@ true)
 		esac
 		;;
 	cygwin*)
-		shrpldflags="$shrpldflags -Wl,--out-implib=libperl.dll.a -Wl,--image-base,0x52000000"
+		shrpldflags="$shrpldflags -Wl,--out-implib=libperl.dll.a"
 		linklibperl="-L. -lperl"
 		;;
 	sunos*)
-- 
2.32.0

