From deac06133585a68f1d91665f3345e0a0b55a09f8 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Tue, 14 Feb 2017 14:45:21 +0300
Subject: [PATCH 4/7] perl.cygwin-Configure-libpth

---
 Configure | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/Configure b/Configure
index 1ea8776..3ae2163 100755
--- a/Configure
+++ b/Configure
@@ -4982,18 +4982,21 @@ esac
 case "$ccname" in
     gcc)
 	$echo 'extern int foo;' > try.c
-	set X `$cppstdin -v try.c 2>&1 | $awk '/^#include </,/^End of search /'|$cppfilter $grep '/include'`
+	set X `$cppstdin -v try.c 2>&1 | $awk '/^#include </,/^End of search /'|$cppfilter $grep '/include$'`
 	shift
 	if $test $# -gt 0; then
 	    incpth="$incpth $*"
 	    incpth="`$echo $incpth|$sed 's/^ //'`"
-	    for i in $*; do
-		j="`$echo $i|$sed 's,/include$,/lib,'`"
-		if $test -d $j; then
-		    libpth="$libpth $j"
-		fi
-	    done
-	    libpth="`$echo $libpth|$sed 's/^ //'`"
+	   for i in $*; do
+	       j="`$echo $i|$sed 's,/include$,/lib,'`"
+	       if $test -d $j; then
+		   case " $libpth " in
+		   *" $j "*) ;;
+		   *) libpth="$libpth $j";;
+		   esac
+	       fi
+	   done
+	   libpth="`$echo $libpth|$sed 's/^ //'`"
 	    for xxx in $libpth $loclibpth $plibpth $glibpth; do
 		if $test -d $xxx; then
 		    case " $libpth " in
-- 
2.32.0

