diff --git a/Makefile.SH b/Makefile.SH
index 5b850f2..6e01fe6 100755
--- a/Makefile.SH
+++ b/Makefile.SH
@@ -358,7 +358,7 @@ define?*)
 	$spitshell >>$Makefile <<!GROK!THIS!
 # Macros to invoke a copy of our fully operational perl during the build.
 PERL_EXE = perl\$(EXE_EXT)
-RUN_PERL = \$(LDLIBPTH) \$(RUN) $perl\$(EXE_EXT)
+RUN_PERL = \$(LDLIBPTH) \$(RUN) ./$perl\$(EXE_EXT)
 !GROK!THIS!
 	;;
 *)
