From 9eb3137f8be9250873c3bf5254eda5baca88450e Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Sat, 2 Nov 2013 12:42:49 +0400
Subject: [PATCH] MSYS2 support for cURL

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 m4/curl-compilers.m4 | 2 +-
 m4/xc-lt-iface.m4    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/m4/curl-compilers.m4 b/m4/curl-compilers.m4
index e732a87..920ef73 100644
--- a/m4/curl-compilers.m4
+++ b/m4/curl-compilers.m4
@@ -921,7 +921,7 @@ AC_DEFUN([CURL_SET_COMPILER_WARNING_OPTS], [
           dnl Only clang 3.2 or later
           if test "$compiler_num" -ge "302"; then
             case $host_os in
-            cygwin* | mingw*)
+            cygwin* | msys* | mingw*)
               dnl skip missing-variable-declarations warnings for cygwin and
               dnl mingw because the libtool wrapper executable causes them
               ;;
diff --git a/m4/xc-lt-iface.m4 b/m4/xc-lt-iface.m4
index 465767f..752c8f9 100644
--- a/m4/xc-lt-iface.m4
+++ b/m4/xc-lt-iface.m4
@@ -263,7 +263,7 @@ elif test "x$allow_undefined_flag" = 'xunsupported'; then
   xc_lt_shlib_use_no_undefined='yes'
 fi
 case $host_os in @%:@ (
-  cygwin* | mingw* | pw32* | cegcc* | os2* | aix*)
+  cygwin* | msys* | mingw* | pw32* | cegcc* | os2* | aix*)
     xc_lt_shlib_use_no_undefined='yes'
     ;;
 esac
